platform/ios/MBXContextSnapshotViewController.m

```diff
@@ -0,0 +1,77 @@
+#import "MBXContextSnapshotViewController.h"
+#import <Mapbox/Mapbox.h>
+
+@interface MBXContextSnapshotViewController () <MGLMapViewDelegate>
+@property MGLMapView *mapView;
+@property UIButton *button;
+@property UIImageView *imageView;
+@end
+
+@implementation MBXContextSnapshotViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    _mapView = [[MGLMapView alloc] initWithFrame:CGRectMake(0, 0, self.view.bounds.size.width, self.view.bounds.size.height / 2) styleURL:[MGLStyle lightStyleURL]];
+    _mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+
+
+    [_mapView setCenterCoordinate:_mapView.centerCoordinate zoomLevel:1 animated:NO];
+    [self.view addSubview:_mapView];
+
+
+    _button = [[UIButton alloc] initWithFrame:CGRectMake(_mapView.frame.origin.x + 100, _mapView.frame.origin.y + 45, 170, 30)];
+    _button.layer.cornerRadius = 15;
+    _button.backgroundColor = [UIColor blueColor];
+    [_button setTitle:@"Create snapshot" forState:UIControlStateNormal];
+    [_button addTarget:self action:@selector(createSnapshot) forControlEvents:UIControlEventTouchUpInside];
+    [self.view addSubview:_button];
+
+    _imageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, self.view.bounds.size.height / 2, self.view.bounds.size.width, self.view.bounds.size.height / 2)];
+    _imageView.backgroundColor = [UIColor blackColor];
+    _imageView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    [self.view addSubview:_imageView];
+}
+
+- (void)createSnapshot {
+
+    MGLMapSnapshotOptions *options = [[MGLMapSnapshotOptions alloc] initWithStyleURL:_mapView.styleURL camera:_mapView.camera size:_mapView.bounds.size];
+
+    options.zoomLevel = _mapView.zoomLevel;
```

Comment:
In most cases, we should prefer to access ivars via their properties, so `self.mapView`, rather than `_mapView`. `-init` and `dealloc` are the exceptions.